### PR TITLE
TokenScheme TIP compliance

### LINKED ISSUE #1269: Update `SimpleTokenScheme` and API
Changes to:
- TIP-18: https://github.com/iotaledger/tips/pull/38/commits/4257b772cab90178832ad7f43fa60d6cdd1ddc85
- TIP-25: https://github.com/iotaledger/tips/pull/57/commits/83dd9f679179b856c2316d4fb9f5d3cff75566dd

@thibault-martinez Will this be part of #977?

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by samuel-rufi — DISMISSED** [review]

**Review by samuel-rufi — APPROVED** [review]

File: bee-api/bee-rest-api/src/types/dtos.rs [hunk 1]
```diff
@@ -1559,9 +1551,67 @@ impl TryFrom<&TokenTagDto> for TokenTag {
...
+    pub maximum_supply: U256Dto,
+}
+
+impl From<&SimpleTokenScheme> for SimpleTokenSchemeDto {
+    fn from(value: &SimpleTokenScheme) -> Self {
+        Self {
+            kind: SimpleTokenScheme::KIND,
+            minted_tokens: U256Dto(value.minted_tokens().to_string()),
+            melted_tokens: U256Dto(value.melted_tokens().to_string()),
+            maximum_supply: U256Dto(value.maximum_supply().to_string()),
```
Comment (grtlr) [inline bee-api/bee-rest-api/src/types/dtos.rs:1599]: I know it's not part of this PR, but do the `U256Dto` fields have the right encoding? For `minted_tokens`, `melted_tokens`, and `maximum_supply` we simply use `to_string` on `U256`. And for `NativeTokens` we use `serde`.

Shouldn't all of these use `prefix_hex`? I might have missed that on my PR.

File: bee-api/bee-rest-api/src/types/dtos.rs [hunk 2]
```diff
@@ -1559,9 +1551,67 @@ impl TryFrom<&TokenTagDto> for TokenTag {
...
+}
+
+impl TryFrom<&SimpleTokenSchemeDto> for SimpleTokenScheme {
+    type Error = Error;
+
+    fn try_from(value: &SimpleTokenSchemeDto) -> Result<Self, Self::Error> {
+        Self::new(
+            U256::from_str(&value.minted_tokens.0).map_err(|_| Error::InvalidField("mintedTokens"))?,
+            U256::from_str(&value.melted_tokens.0).map_err(|_| Error::InvalidField("meltedTokens"))?,
+            U256::from_str(&value.maximum_supply.0).map_err(|_| Error::InvalidField("maximumSupply"))?,
```
Comment (grtlr) [inline bee-api/bee-rest-api/src/types/dtos.rs:1611]: Same as above, use `prefix_hex`?

File: bee-api/bee-rest-api/src/types/dtos.rs (hunk 1 above)
Comment (thibault-martinez) [reply bee-api/bee-rest-api/src/types/dtos.rs:1599]: May I ask you to change this in a a subsequent PR ?

Comment (grtlr) [reply bee-api/bee-rest-api/src/types/dtos.rs:1599]: Sure, no worries.

**Review by grtlr — APPROVED** [review]

**Pull request merged**